### DISCUSSION TITLE
Values clause support in HQL/Criteria

### DISCUSSION THREAD

**Discussion by beikov** [hibernate/hibernate-orm#7067] [Design Proposals]
Currently we only support the VALUES clause in insert statements, but it might be nice to support this anywhere in HQL where a subquery from clause item is allowed. Use cases range from emulating `generate_series` up to some reporting where people want to join fixed values like e.g. time periods against aggregated data. I propose the following syntax:

```
values_from_item
    : '(' VALUES values_item ( ',' values_item  )* ')' (AS? alias column_aliases?)?
    ;
values_item : '(' expression ( ',' expression )* ')' ;
value_count : INTEGER;
column_aliases : '(' identifier (',' identifier)* ')'
```

**Comment by dreab8** [comment]
https://hibernate.atlassian.net/browse/HHH-19403
